apollo-router/src/configuration/metrics.rs

```diff
@@ -360,6 +362,25 @@ impl InstrumentData {
         self.data
             .insert("apollo.router.env".to_string(), (1, attributes));
     }
+
+    pub(crate) fn populate_license_instrument(&mut self, license_state: &LicenseState) {
+        self.data.insert(
+            "apollo.router.license".to_string(),
+            (
+                1,
+                [(
+                    "license.state".to_string(),
+                    match license_state {
+                        LicenseState::Licensed => "licensed".into(),
+                        LicenseState::LicensedWarn => "warn".into(),
+                        LicenseState::LicensedHalt => "halt".into(),
+                        LicenseState::Unlicensed => "unlicensed".into(),
+                    },
```

Comment:
We should implement `Display` on `LicenseState`, it might be useful later for another usecase

Reply:
I had to rebase because the root banch changed, but I did this in [78943d6](https://github.com/apollographql/router/pull/4365/commits/78943d60c0f218892ebafa66b796a14b1f4d6d69)